### PR TITLE
hack verify.sh: clean up "base"

### PR DESCRIPTION
**Pull request by skitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This is a follow-up for commit beb51e1 ("make verify: run checks in
all module hack directories"). Checking only modules known to go ends
up ignoring modules which aren't dependencies, such as
./staging/src/k8s.io/code-generator/examples/hack; but the checks
should be run there too. So this finds modules by looking for go.mod
files.

Use PWD in run-checks to record failing verification scripts instead
of a hidden dependency on an externally-set variable.

#### Which issue(s) this PR fixes:
Fixes #

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/make-rules/verify.sh [hunk 1]
```diff
@@ -234,10 +234,9 @@ ret=0
 modules=() # Pacify shellcheck
 kube::util::read-array modules < <(go list -f '{{.Dir}}' -m)
```
Comment (aojea) [inline hack/make-rules/verify.sh:235]: I just realize this omits these two
```
./staging/src/k8s.io/code-generator/examples/hack
./staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack
```

do we need them?

Comment (skitt) [reply hack/make-rules/verify.sh:235]: Oops, nice catch — what prompted me to do all this initially is `./staging/src/k8s.io/code-generator/examples/hack`, so yes, we do need them!

**Comment by aojea** [comment]
/lgtm
/assign @thockin 

it seems now it runs the examples

```
diffing /home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/code-generator/examples against freshly generated codegen
/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/code-generator/examples up to date.
Smoke testing examples by compiling...
/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/code-generator/examples /home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/code-generator/examples
/home/prow/go/src/k8s.io/kubernetes/staging/src/k8s.io/code-generator/examples
+++ exit code: 0
```

**Pull request merged**